samcli/hook_packages/terraform/hooks/prepare.py

```diff
@@ -93,6 +108,29 @@ def prepare(params: dict) -> dict:
         raise PrepareHookException(f"Unable to create directory {output_dir_path}") from e
 
 
+def _update_resources_paths(cfn_resources: Dict[str, Any], terraform_application_dir: str) -> None:
+    """
+    As Sam Cli and terraform handles the relative paths differently. Sam Cli handles the relative paths to be relative
+    to the template, but terraform handles them to be relative to the project root directory. This Function purpose is
+    to update the CFN resources paths to be absolute paths, and change relative paths to be relative to the terraform
+    application root directory.
+
+    Parameters
+    ----------
+    cfn_resources: dict
+        CloudFormation resources
+    terraform_application_dir: str
+        The terraform application root directory where all paths will be relative to it
+    """
+    resources_attributes_to_be_updated = {CFN_AWS_LAMBDA_FUNCTION: ["Code"]}
+    for _, resource in cfn_resources.items():
+        if resource.get("Type") in resources_attributes_to_be_updated:
+            for attribute in resources_attributes_to_be_updated[resource["Type"]]:
+                original_path = resource["Properties"].get(attribute)
```

Comment:
it should not be a valid case, as we generate these resources, but I can update the code to avoid any unexpected issues.

Reply:
This might be a good assumption now but might not be in the future. Some resources do not require Properties to be set. 

Instead, we should be doing defensive programming for these things. Otherwise they will lead to crashes on the cli.

Reply:
I already updated the code.

Reply:
O it is baked in line 128. I expected this to be updated to `resources.get("Properties", {})` instead.

Reply:
`isinstance(resource.get("Properties"), dict)` condition should prevent us reaching this point if  resources.get("Properties") is None, but any way, I will update it.

Reply:
Done